File: 04-vector-search/python/SearchComponents/completion.py
import json

def generate_completion(prompt,completion_deployment,AzureOpenAICompletionClient,user_input):
    system_prompt = '''
    You are an intelligent assistant for the Adventure Works Bike Shop.
    You are designed to provide helpful answers to user questions about the store invetory given the information about to be provided.
        - Only answer questions related to the information provided below, provide 3 clear suggestions in a list format.
        - Write two lines of whitespace between each answer in the list.
        - Only provide answers that have products that are part of the Adventure Works Bike Shop.
        - If you're unsure of an answer, you can say ""I don't know"" or ""I'm not sure"" and recommend users search themselves."
    '''

    messages=[
        {"role": "system", "content": system_prompt},
        {"role": "user", "content": user_input},
    ]
    
    for item in prompt:
        messages.append({"role": "system", "content": item['document']['categoryName']+" "+item['document']['name']})
    response = AzureOpenAICompletionClient.chat.completions.create(model=completion_deployment, messages=messages)
    completions = json.loads(response.model_dump_json(indent=2))

    return completions
    

File: 04-vector-search/node.js/SearchComponents/completion.js
const generateCompletion = async (prompt, completionDeployment, AzureOpenAICompletionClient, userInput) => {
    const systemPrompt = `
    You are an intelligent assistant for the Adventure Works Bike Shop.
    You are designed to provide helpful answers to user questions about the store inventory given the information about to be provided.
        - Only answer questions related to the information provided below, provide 3 clear suggestions in a list format.
        - Write two lines of whitespace between each answer in the list.
        - Only provide answers that have products that are part of the Adventure Works Bike Shop.
        - If you're unsure of an answer, you can say "I don't know" or "I'm not sure" and recommend users search themselves.
    `;

    let messages = [
        {role: "system", content: systemPrompt},
        {role: "user", content: userInput},
    ];

    for (let item of prompt) {
        messages.push({role: "system", content: `${item.document.categoryName} ${item.document.name}`});
    }

    const response = await AzureOpenAICompletionClient.getChatCompletions(completionDeployment,messages);
    
    return response;
}

module.exports.generateCompletion = generateCompletion;